src/bvals/cc/bnd_info.cpp

```diff
@@ -338,6 +338,18 @@ BndInfo BndInfo::GetSetBndInfo(std::shared_ptr<MeshBlock> pmb, const NeighborBlo
                             pmb.get());
     out.var = v->data.Get();
   }
+
+  if (buf->GetState() == BufferState::received) {
+    // With control variables, we can end up in a state where a
+    // variable that is not receiving null data is unallocated.
+    // for allocated to be set, the buffer must be sending non-null
+    // data and the receiving variable must be allocated
+    out.allocated = v->IsAllocated();
+  } else if (buf->GetState() == BufferState::received_null) {
```

Comment:
I don't think that the buffer can change state in between those two calls since everything is serial on a particular rank. I switched though to possibly future proof (i.e. if there are multiple threads), although I think there may be other race conditions if the buffer states are being updated by multiple threads.